src/crlf.c

```diff
@@ -147,11 +150,28 @@ static int crlf_apply_to_odb(
 			switch (ca->safe_crlf) {
 			case GIT_SAFE_CRLF_FAIL:
 				giterr_set(
-					GITERR_FILTER, "LF would be replaced by CRLF in '%s'",
-					git_filter_source_path(src));
+					GITERR_FILTER, lfwarning, git_filter_source_path(src));
 				return -1;
 			case GIT_SAFE_CRLF_WARN:
-				/* TODO: issue warning when warning API is available */;
+			{
+				int ret;
+				git_warning_crlf warning;
+				git_buf buf = GIT_BUF_INIT;
+
+				if (git_buf_printf(&buf, lfwarning, git_filter_source_path(src)) < 0)
```

Comment:
I bet this kind of stuff would come up quite a lot. So a function to build a warning with a type and a format string would be nice, as well as one cleaning up the warning